### ISSUE TITLE
calculate model metrics

### ISSUE THREAD

**Issue by drbenvincent** [drbenvincent/darc-analysis-python#3]
All these models are basically doing binary classification of responses. So we should calculate some metrics after sampling. Appropriate metrics are:
- [x] Log Loss: [sklearn.metrics.log_loss](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html)
- [x] Area Under Curve: [sklearn.metrics.roc_auc_score](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score)

Note: We'll get a distribution of values back for each metric, one for each sample.

Computing these in the PyMC3 model itself I don't think is going to work because that all needs to be theano compatible? So I'm going with doing the metric calculations outside of the model, after sampling. This does mean that I'll need to think about how to store/present/plot that information. But for the moment I'll just pop all these values in a dictionary and print the median metric values.

- [x] Bear in mind that there are some paradigms where continuous responses are given, so it might be good to have a `BinaryResponseModel` subclass of model.

**Extracting the values**
If you have a single model object, then you can do something like:

    auc_median = np.median(model.metrics['auc'])

of if you have a list of models, then:

    auc_median = [np.median(model.metrics['auc']) for model in models]
